### DISCUSSION TITLE
Any questions? comments? observations?

### DISCUSSION THREAD

**Discussion by mmcc-xx** [mmcc-xx/WhosAtMyFeeder#1] [General]
Please feel free to share.

**Comment by FrigateEnthusiast** [comment]
Love it and will soon give it a try. What area of the world/US are the bird id's from? 

Maybe I'll add an enhancement/add-in to detect squirrels, then aim a 40W laser on a 2 axis mount to the center of the bounding box, then fire off a few quick bursts. lol

**Comment by mmcc-xx** [comment]
I don't know what area the bird IDs are from, but it seems to be working on the birds in northeast Ohio pretty well

This is the model: https://tfhub.dev/google/lite-model/aiy/vision/classifier/birds_V1/3
It says it is using the iNaturalist dataset. I don't know more than that.

Regarding squirrels, do I ever have good new for you. I have recently closed the loop on the squirrels that come to my feeders.
https://www.reddit.com/r/homeautomation/comments/13bn798/behold_the_power_of_the_squirrelinator_warning/

I need to put together some details on how this was done, but Frigate is doing the detecting. Even though there is no "squirrel" in the Firgate model, I lumped together all of the furry mammals in the model into one label called "animal" in the Frigate config.yml file like so:
```
model:
  labelmap:
    16: animal #cat
    17: animal #dog
    18: animal #horse
    19: animal #sheep
    22: animal #bear
```

That does a pretty good job at catching squirrels and racoons. Also my dogs, but they can handle getting sprayed.

No 2 axis mount for aiming.... yet.

**Comment by tusspot88** [comment]
I'm setting this up on Portrainer on a Debian VM. I'm a bit of a novice on the Linux side.

1. I can't map a file directly with Portrainer (e.g. the species.db) if I'm reading the documentation correctly. Any alternatives to do this other than not persisting the database?
2. My MQTT install requires credentials. I don't see any notes about supporting a username or password this in the config file. Do you have any tricks for this?

**Reply by mmcc-xx** [reply]
Regarding #1, I've never used portainer, so I'm not really sure how it works. I'll poke around a little bit and see how that would work.

Regarding #2, I added an option for MQTT authentication. I think it will work :) Please let me know.

**Reply by mmcc-xx** [reply]
Regarding #1, I made a directory called "data" to house the database, so you can bind to that directory instead of the file directly.

**Reply by tusspot88** [reply]
I reinstalled it on my instance and it's looking great!

Can it handle multiple cameras / MQTT topics if I format the config.yml correctly?

I have two cameras pointed at birdfeeders currently, and considering a third.

**Reply by mmcc-xx** [reply]
Well it didn't happen yesterday, but I think I have multiple cameras working. I don't have multiple cameras to test it with, so please give it a go. You'll need to update the config file and make the "camera:" item an array. The config file in the repo is updated - just add one camera name per line like...
```
  camera:
    - birdcam
    - othercamera
    - yetanothercamera
```
Please let me know if it works.

**Reply by tusspot88** [reply]
I loaded it and it looks like it's working. My second feeder near the garden doesn't get as much attention, so it might be a while until it catches something.
My third feeder is on an upstairs porch and gets a ton of attention, but I don't have a camera for that one yet.

**Comment by mmcc-xx** [comment]
Awesome! Single camera currently but it’ll be easy to add support for
multiple (I think) - just a matter of subscribing to multiple camera
snapshot events. I’ll work on it later today.

**Comment by tusspot88** [comment]
I noticed the detail on the common names has changed. I'm seeing less names, and for some of the common birds around my parts (bluebirds and goldfinches) I'm only seeing the scientific name now. Did something change or did I mess something up?
![image](https://github.com/mmcc-xx/WhosAtMyFeeder/assets/107322550/9f4c76b9-82aa-4b2a-b04d-47c7d35b230f)

**Reply by mmcc-xx** [reply]
Huh - that’s odd. Are you using the image from Docker Hub or building your own?

**Reply by tusspot88** [reply]
I'm pulling it from Docker Hub.

    `image: mmcc73/whosatmyfeeder:latest`

**Reply by mmcc-xx** [reply]
I'm using this library called EcoNameTranslator to get the common names. Since it is pretty slow, when it gets a common name I store it in the local database to use later. I suspect that the call to EcoNameTranslator failed for some reason (maybe the container didn't have internet access?) and my code ended up saving the scientific name in the common name field. I'll take a look.

**Reply by mmcc-xx** [reply]
I added a static database of bird names to look up based on data from eBird. It looks like it is missing a few common ones or their taxonomy changed or something, but overall works ok and is fast. Pushing the new image now.

**Comment by scy59** [comment]
Any plans to make a Home Assistant OS add-on? Maybe fire off MQTT messages on species detection? 

I'd thank you, but my hungry visiting ducks would thank you more.

**Reply by linkedupbits** [reply]
I would also be keen to see an MQTT push on identification (for Home Assistant, but MQTT could be agnostic) to get live notifications when particular species arrive…

**Reply by mmcc-xx** [reply]
Hi - thanks for the input. Yes that's something I want to do. I've mostly been working on the bird-id-by-sound app lately (https://github.com/mmcc-xx/BirdCAGE) but I want to build in notifications and HA integration for both (and, I think, somehow integrate the two as the source for audio and video will often be the same device.).

Clark

**Reply by linkedupbits** [reply]
Would this be an approach for higher quality matches with the tflite model:

- mount the Frigate media folder (read only) into the docker container 
- use MQTT to be notified when a bird is recognised
- Use the high quality snapshot from Frigate vs. the low res MQTT jpeg to identify the species
Just a thought - I haven’t looked at your source code!

Your solution is definitely working though!

**Comment by ZogKarndon** [comment]
I just set my instance up, and while I am getting bird events from Frigate, I'm not getting any events from WhosAtMyFeeder,  Are there some logs I could look at?  I can't seem to find any.

**Reply by mmcc-xx** [reply]
You can look at the logs from docker by running `docker logs whosatmyfeeder` or whatever the name of your container is. 

You should see entries that looks like this:
```
Category(index=320, score=0.31640625, display_name='Haemorhous mexicanus', category_name='/m/0206yg')
2023-06-11 18:47:01
Category(index=320, score=0.4921875, display_name='Haemorhous mexicanus', category_name='/m/0206yg')
2023-06-11 18:47:09
Category(index=320, score=0.25390625, display_name='Haemorhous mexicanus', category_name='/m/0206yg')
2023-06-11 18:47:19
Category(index=650, score=0.95703125, display_name='Melospiza melodia', category_name='/m/01y5dx')
Storing...
```

The score has to be greater than the threshold value in the config file for it to get flagged as a detection, so that is something to check.
